cpu/stm32l1/periph/i2c.c

```diff
@@ -353,12 +376,30 @@ void i2c_poweroff(i2c_t dev)
     }
 }
 
-static void _start(I2C_TypeDef *i2c, uint8_t address, uint8_t rw_flag)
+static void _i2c_reset(I2C_TypeDef *i2c) {
+    DEBUG("I2C: Resetting the bus\n");           
+    
+    uint16_t ccr = i2c->CCR;
+
+    i2c->CR1 |= I2C_CR1_SWRST;
+    i2c->CR1 &= ~I2C_CR1_SWRST;           
+    
+    _i2c_init(i2c, ccr);
+}
+
+static int _start(I2C_TypeDef *i2c, uint8_t address, uint8_t rw_flag)
 {
     /* wait for device to be ready */
     DEBUG("Wait for device to be ready\n");
 
-    while (i2c->SR2 & I2C_SR2_BUSY) {}
+    uint32_t time_now = xtimer_now_usec();
+    while (i2c->SR2 & I2C_SR2_BUSY) {
+        /* 100 ms timeout */
```

Comment:
why 100ms? I could not find anything in the standard (though I did no look very closely...). So if this is simply a value that turned out to be a good practice, I think it makes sense to a) document it, and/or b) make it configurable (I guess through either `periph_conf.h` or `periph_cpu.h`).